include/aspect/utilities.h

```diff
@@ -411,6 +427,44 @@ namespace aspect
       return std::vector<T> ();
     }
 
+    template <int dim>
+    std::vector<std::string>
+    expand_dimensional_variable_names (const std::vector<std::string> &var_declarations)
```

Comment:
I think the implementation of this might as well go into the `.cc` file. You'll have to add instantiations to the bottom, though.